dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java

```diff
@@ -0,0 +1,103 @@
+import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link HttpServerHandler} deals with HTTP requests received from Netty Channel.
+ */
+public class HttpServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+  private final PagedService mPagedService;
+
+  /**
+   * {@link HttpServerHandler} deals with HTTP requests received from Netty Channel.
+   * @param pagedService the {@link PagedService} object provides page related RESTful API
+   */
+  public HttpServerHandler(PagedService pagedService) {
+    mPagedService = pagedService;
+  }
+
+  @Override
+  public void channelReadComplete(ChannelHandlerContext ctx) {
+    ctx.flush();
+  }
+
+  @Override
+  public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
+    if (msg instanceof HttpRequest) {
+      HttpRequest req = (HttpRequest) msg;
+      String requestUri = req.uri();
+      // parse the request uri to get the parameters
+      // TODO(JiamingMai): parse the URI and dispatch it to different methods
+      requestUri = requestUri.substring(requestUri.indexOf("?") + 1);
+      String[] params = requestUri.split("&");
+      Map<String, String> parametersMap = new HashMap<>();
+      for (String param : params) {
+        String[] keyValue = param.split("=");
+        parametersMap.put(keyValue[0], keyValue[1]);
+      }
+      String fileId = parametersMap.get("fileId");
+      long pageIndex = Long.parseLong(parametersMap.get("pageIndex"));
+
+      boolean keepAlive = HttpUtil.isKeepAlive(req);
+      ByteBuf byteBuf = mPagedService.getPage(fileId, pageIndex, ctx.channel());
+      FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), OK,
+          byteBuf);
+      response.headers()
+          .set(CONTENT_TYPE, TEXT_PLAIN)
+          .setInt(CONTENT_LENGTH, response.content().readableBytes());
+
+      if (keepAlive) {
+        if (!req.protocolVersion().isKeepAliveDefault()) {
+          response.headers().set(CONNECTION, KEEP_ALIVE);
+        }
+      } else {
+        // Tell the client we're going to close the connection.
+        response.headers().set(CONNECTION, CLOSE);
+      }
+
+      ChannelFuture f = ctx.write(response);
```

Comment:
```suggestion
    ctx.write(response);
    ctx.write(new DefaultFileRegion(xxxxxx);
    ChannelFuture future = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
```
you can try this way to write defaultFileRegion to the channel.

Reply:
Good idea. I am trying implementing the server by returning `DefaultFileRegion`.